src/spatialdata/_core/spatialdata.py

```diff
@@ -312,24 +312,23 @@ def _init_add_element(self, name: str, element_type: str, overwrite: bool) -> za
             return elem_group
         return root
 
-    def _locate_spatial_element(self, element: SpatialElement) -> tuple[str, str]:
+    def locate_element(self, element: SpatialElement) -> tuple[str, str] | None:
         """
-        Find the SpatialElement within the SpatialData object.
+        Return the name and the type of a SpatialElement within the SpatialData object.
 
         Parameters
         ----------
         element
             The queried SpatialElement
 
-
         Returns
         -------
-        name and type of the element
+        name and type of the element; if the element is not found, None is returned instead
 
         Raises
         ------
         ValueError
-            the element is not found or found multiple times in the SpatialData object
+            the element is found multiple times in the SpatialData object
```

Comment:
and, precisely because there is nothing wrong about storing 2 exact same images in 2 different keys, that such odd behaviors should be left to the users and not being a method provided in the API imho

Reply:
I would discourage it because as @aeisenbarth pointed out it could lead to weird behavior downstream. Here I do the check so we ensure that the return path is non-ambiguous.

Reply:
Also Wouter's comment here: https://github.com/scverse/spatialdata/issues/425#issuecomment-1883408895

Reply:
sorry @LucaMarconato but I think the amount of complexity added on checking the consistency is tbh a lot more than just returning 2 strings here. I think a warning for sure is necessary here, but not a fail, as it should fail earlier.

Reply:
I was thinking more of returning the first string "trusting" that the object don't contain a copy. I really think that we should not allow at this stage to have duplicate elements. Maybe we could add it later in a separate PR after deep testing, but right now it would the make the library less stable without a real use case.